skimage/exposure/tests/test_exposure.py

```diff
@@ -12,30 +12,109 @@
 from skimage._shared import testing
 from skimage._shared.testing import (assert_array_equal,
                                      assert_array_almost_equal,
+                                     assert_equal,
                                      assert_almost_equal)
 
 
 # Test integer histograms
 # =======================
 
+def test_wrong_source_range():
+    im = np.array([-1, 100], dtype=np.int8)
+    with testing.raises(ValueError):
+        frequencies, bin_centers = exposure.histogram(im, source_range='foobar')
+
+
 def test_negative_overflow():
-    im = np.array([-1, 127], dtype=np.int8)
-    frequencies, bin_centers = exposure.histogram(im)
-    assert_array_equal(bin_centers, np.arange(-1, 128))
+    im = np.array([-1, 100], dtype=np.int8)
+    frequencies, bin_centers = exposure.histogram(im, source_range='image')
+    assert_array_equal(bin_centers, np.arange(-1, 101))
     assert frequencies[0] == 1
     assert frequencies[-1] == 1
     assert_array_equal(frequencies[1:-1], 0)
 
 
 def test_all_negative_image():
-    im = np.array([-128, -1], dtype=np.int8)
-    frequencies, bin_centers = exposure.histogram(im)
-    assert_array_equal(bin_centers, np.arange(-128, 0))
+    im = np.array([-100, -1], dtype=np.int8)
+    frequencies, bin_centers = exposure.histogram(im, source_range='image')
```

Comment:
why do you set explicitly ``source_range`` to 'image' if it's the default?

Reply:
This is just to ease the reading.

Reply:
ok, but we should have some of the tests which fail if we decide to change the default in the future, therefore it's preferable not to specify the kw argument in at least some of the tests.

Reply:
@emmanuelle it's done.